src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -236,4 +227,26 @@ private static boolean checkIsAllLinesAreWhitespace(String[] lines, int lineFrom
         }
         return result;
     }
+
+    /**
+     * Calculates the left curly corresponding to the block to be checked.
+     *
+     * @param ast a {@code DetailAST} value
+     * @return the left curly corresponding to the block to be checked
+     */
+    private static DetailAST findLeftCurly(DetailAST ast) {
+        DetailAST leftCurly;
+        if (ast.findFirstToken(TokenTypes.SLIST) == null) {
+            leftCurly = ast.findFirstToken(TokenTypes.LCURLY);
+        }
+        else {
+            leftCurly = ast.findFirstToken(TokenTypes.SLIST);
+        }
+        if (ast.getType() == TokenTypes.LITERAL_CASE
+                && ast.getNextSibling() != null
+                && "{".equals(ast.getNextSibling().getFirstChild().getText())) {
```

Comment:
such code `"{".equals .........` is not accepted, you should do all by Tokens like `ast.findFirstToken(TokenTypes.LCURLY);`

if CASE is so special than IF for it should be the first in method to do previous logic only if ast is not a CASE.... please prove that CASE AST is so special.